frame/identity/src/lib.rs

```diff
@@ -451,9 +456,112 @@ decl_error! {
 		TooManyFields,
+		/// Maximum amount of registrars reached. Cannot add any more.
+		TooManyRegistrars,
 }
 }
 
+/// Functions for calcuating the weight of dispatchables.
+mod weight_for {
+	use frame_support::weights::{RuntimeDbWeight, Weight};
+
+	/// Weight calculation for `set_identity`.
+	pub(crate) fn set_identity(
+		db: RuntimeDbWeight,
+		judgements: impl Into<Weight>,
+		extra_fields: impl Into<Weight>
+	) -> Weight {
+		db.reads_writes(1, 1)
+			+ db.reads_writes(1, 1) // balance ops
+			+ 150_000_000 // constant
+			+ 700_000 * judgements.into() // R
+			+ 3_000_000 * extra_fields.into() // X
+	}
+
+	/// Weight calculation for `set_subs`.
+	pub(crate) fn set_subs(
+		db: RuntimeDbWeight,
+		old_subs: impl Into<Weight> + Copy,
+		subs: impl Into<Weight> + Copy
+	) -> Weight {
+		db.reads(1) // storage-exists (`IdentityOf::contains_key`)
+			+ db.reads_writes(1, 1) // balance ops
+			+ db.reads_writes(1, old_subs.into()) // P old DB deletions
+			+ db.writes(subs.into() + 1) // S + 1 new DB writes
+			+ 130_000_000 // constant
+			+ 5_200_000 * old_subs.into() // P
+			+ 7_300_000 * subs.into() // S
+	}
+
+	/// Weight calculation for `clear_identity`.
+	pub(crate) fn clear_identity(
+		db: RuntimeDbWeight,
+		judgements: impl Into<Weight>,
+		subs: impl Into<Weight> + Copy,
+		extra_fields: impl Into<Weight>
+	) -> Weight {
+		db.reads_writes(2, subs.into() + 2) // S + 2 deletions
+			+ db.reads_writes(1, 1) // balance ops
+			+ 160_000_000 // constant
+			+ 500_000 * judgements.into() // R
+			+ 5_400_000 * subs.into() // S
+			+ 2_000_000 * extra_fields.into() // X
+	}
+
+	/// Weight calculation for `request_judgement`.
+	pub(crate) fn request_judgement(
+		db: RuntimeDbWeight,
+		judgements: impl Into<Weight>,
+		extra_fields: impl Into<Weight>
+	) -> Weight {
+		db.reads_writes(2, 1)
+			+ db.reads_writes(1, 1) // balance ops
+			+ 180_000_000 // constant
+			+ 950_000 * judgements.into() // R
+			+ 3_400_000 * extra_fields.into() // X
+	}
+
+	/// Weight calculation for `cancel_request`.
+	pub(crate) fn cancel_request(
+		db: RuntimeDbWeight,
+		judgements: impl Into<Weight>,
+		extra_fields: impl Into<Weight>
+	) -> Weight {
+		db.reads_writes(1, 1)
+			+ db.reads_writes(1, 1) // balance ops
+			+ 150_000_000 // constant
+			+ 600_000 * judgements.into() // R
+			+ 3_600_000 * extra_fields.into() // X
+	}
+
+	/// Weight calculation for `provide_judgement`.
+	pub(crate) fn provide_judgement(
+		db: RuntimeDbWeight,
+		judgements: impl Into<Weight>,
+		extra_fields: impl Into<Weight>
+	) -> Weight {
+		db.reads_writes(2, 1)
+			+ 120_000_000 // constant
+			+ 1_100_000 * judgements.into() // R
+			+ 3_500_000 * extra_fields.into()// X
+	}
+
+	/// Weight calculation for `kill_identity`.
+	pub(crate) fn kill_identity(
+		db: RuntimeDbWeight,
+		judgements: impl Into<Weight>,
+		subs: impl Into<Weight> + Copy,
+		extra_fields: impl Into<Weight>
+	) -> Weight {
+		db.reads_writes(3, subs.into() + 3) // 2 `take`s + S deletions
```

Comment:
This should be 2 instead of 3?

Reply:
This might be the treasury ops you noted above?

Reply:
Not sure, but this should be 2 for both read and write

Reply:
yeah, you're right